File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) <= 2:
            return max(nums)
        memorize = [0]*len(nums)
        memorize[-1] = nums[-1]
        memorize[-2] = nums[-2]
        for i in range(len(nums)-3, -1, -1):
            memorize[i] = max(nums[i], nums[i] + max(memorize[i+2:]))
        return max(memorize[0], memorize[1])

File: 0198-house-robber/0198-house-robber.go
func rob(nums []int) int {
    dp := make([]int, len(nums))
    dp[0] = nums[0]
    for i := 1; i < len(nums); i++ {
        if i == 1 {
            dp[i] = max(nums[0], nums[1])
        } else {
            dp[i] = max(dp[i-1], dp[i-2] + nums[i])
        }
    }

    return dp[len(dp)-1]
}